addons/mail/static/src/models/chat_window.js

```diff
@@ -158,6 +158,36 @@ registerModel({
                 this.update({ isFocused: true });
             }
         },
+        /**
+         * Save the scroll positions of the chat window in the store.
+         * This is useful in order to remount chat windows and keep previous
+         * scroll positions. This is necessary because when toggling on/off
+         * home menu, the chat windows have to be remade from scratch.
+         */
+        saveThreadScrollTop() {
+            if (
+                !this.threadView ||
+                !this.threadView.messageListView ||
+                !this.threadView.messageListView.component ||
+                !this.threadViewer
+            ) {
```

Comment:
I would also test `this.exists()` to be consistent with `!this.chatWindow`.

Reply:
I moved the `!this.chatWindow` one level above, in `_onClickHeader`, so that there's the guard specifically on the handler method.

That's why I could simply remove `!this.exists()` here.

Reply:
I wouldn't. It's best for code to guard the earliest in flow, so that you don't have to guard at all corners.
The guard has been moved to handler method, which is the earliest so it's good.